---
title: "Over-representation of IRE genes in MSigDB gene sets - Zebrafish"
author: "Nhi Hin"
date: "2020-04-22"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r load-packages, include=FALSE}
# Load packages
library(dplyr)
library(readr)
library(magrittr)
library(limma)
library(here)
library(openxlsx)
library(org.Dr.eg.db)
library(ggplot2)
library(stringr)
library(purrr)
library(export)
library(reshape2)
library(fgsea)
library(tibble)
library(here)

# ggplot2 theme
theme_set(theme_bw())
```

## 0. Introduction

- Aim of this analysis is to determine the functional significance of the IRE 
gene sets defined for zebrafish. 

- To do this, we will test for over-representation of the IRE gene sets in 
existing gene sets of MSigDB using Fisher's exact test.

- The gene set collections that are relevant to our analysis are:

    - **Hallmark gene sets** (**H**) are coherently expressed signatures 
    derived by aggregating many MSigDB gene sets to represent well-defined
    biological states or processes.

    - **Curated gene sets** (**C2**) are from online pathway databases, 
    publications in PubMed, and knowledge of domain experts.

    - **Motif gene sets** (**C3**) are based on conserved cis-regulatory 
    motifs from a comparative analysis of the human, mouse, rat, and 
    dog genomes.

    - **Gene ontology gene sets** (**C5**) consist of genes 
    annotated by the same GO terms. 

## 1. Import data

- Import in the relevant gene sets below:
    
```{r}
# Relevant gene set collections
h_mapped <- readRDS(here("data", "gsea", "zebrafish",
                         "ens_h_mapped.rds"))  # Hallmark Gene set collection 
c2_mapped <- readRDS(here("data", "gsea", "zebrafish",
                          "ens_c2_mapped.rds"))  # Curated Gene set collection
c3_mapped <- readRDS(here("data", "gsea", "zebrafish",
                          "ens_c3_mapped.rds"))  # Motif Gene set collection
c5_mapped <- readRDS(here("data", "gsea", "zebrafish",
                          "ens_c5_mapped.rds"))  # Gene Ontology Gene set collection 

# IRE gene sets defined for zebrafish
ireGenes <- readRDS(here("data","ire", "zebrafish","ireGenes.rds")) 
```

- Put all important MSigDB gene set collections into a data.frame.

```{r}
h_tib <- h_mapped %>% tibble %>% set_colnames(c("ids")) %>%
  mutate(geneset = names(h_mapped), source = "h") 
c2_tib <- c2_mapped %>% tibble %>% set_colnames(c("ids")) %>%
  mutate(geneset = names(c2_mapped), source = "c2")
c3_tib <- c3_mapped %>% tibble %>% set_colnames(c("ids")) %>% 
  mutate(geneset = names(c3_mapped), source = "c3")
c5_tib <- c5_mapped %>% tibble %>% set_colnames(c("ids")) %>%
  mutate(geneset = names(c5_mapped), source = "c5")

gs <- bind_rows(h_tib, c2_tib, c3_tib, c5_tib)

gs
```

- We will be testing for over-representation of the following IRE gene sets:

    - **All predicted 3' IRE genes (All 3' IRE)**
    - **All predicted 5' IRE genes (All 5' IRE)**

```{r}
allIREGenes <- c(ireGenes$ire3_all, ireGenes$ire5_all)
allIREGenes %>% head
```


## 2. Compute gene set overlap with IRE genesets

- We wish to see the overlap between the genesets in `gs` with the 
`ireGenes` (including 3' and 5' IRE genes), 
as well as separately, with the 3' IRE genes and 5' IRE genes. 
We will add this information into the `gs` tibble.

```{r}
zebrafishGenes <- as.data.frame(org.Dr.egENSEMBL)$ensembl_id

gs %<>% rowwise() %>% mutate(
  # Number of genes in the geneset
  n = length(ids),  
  # Number of genes in the gene set which have 3' or 5' predicted IREs 
  n_with_ire = sum(ids %in% allIREGenes),  
  n_without_ire = (n - n_with_ire),
  universe_with_ire = sum(zebrafishGenes %in% allIREGenes & 
                            !(zebrafishGenes %in% ids)), 
  universe_without_ire = (length(zebrafishGenes) - universe_with_ire),
  
  # Number of genes in the gene set which have 3' predicted IREs
  n_with_ire3 = sum(ids %in% ireGenes$ire3_all),  
  n_without_ire3= (n - n_with_ire),
  universe_with_ire3 = sum(zebrafishGenes %in% ireGenes$ire3_all & 
                             !(zebrafishGenes %in% ids)), 
  universe_without_ire3 = (length(zebrafishGenes) - universe_with_ire3),
  
  # Number of genes in the gene set which have 5' predicted IREs
  n_with_ire5 = sum(ids %in% ireGenes$ire5_all),  
  n_without_ire5= (n - n_with_ire),
  universe_with_ire5 = sum(zebrafishGenes %in% ireGenes$ire5_all & 
                             !(zebrafishGenes %in% ids)), 
  universe_without_ire5 = (length(zebrafishGenes) - universe_with_ire5)
) %>% 
  ungroup()

gs
```

## 3. Contingency tables

- Create contingency table for each gene set. We will store this in a 
list of matrices `gs_mat`. The 3' and 5' contingency matrices are 
stored in `gs_mat3` and `gs_mat5`. 

```{r}
gs_mat <- gs %>% 
  dplyr::select(n_with_ire,
                n_without_ire, 
                universe_with_ire, 
                universe_without_ire) %>% 
  apply(X = ., MARGIN = 1, FUN = function(x){
    x %>% 
      matrix(2,2) %>% 
      t %>% 
      list()
  }) %>% set_names(gs$geneset) %>% 
  lapply(function(x){
    x %>% .[[1]]
  })

gs_mat3 <- gs %>% 
  dplyr::select(n_with_ire3,
                n_without_ire3, 
                universe_with_ire3, 
                universe_without_ire3) %>% 
  apply(X = ., MARGIN = 1, FUN = function(x){
    x %>% 
      matrix(2,2) %>% 
      t %>% 
      list()
  }) %>% set_names(gs$geneset) %>% 
  lapply(function(x){
    x %>% .[[1]]
  })

gs_mat5 <- gs %>% 
  dplyr::select(n_with_ire5,
                n_without_ire5, 
                universe_with_ire5, 
                universe_without_ire5) %>% 
  apply(X = ., MARGIN = 1, FUN = function(x){
    x %>% 
      matrix(2,2) %>% 
      t %>% 
      list()
  }) %>% set_names(gs$geneset) %>% 
  lapply(function(x){
    x %>% .[[1]]
  })

```
```{r}
gs_mat[1:3]
gs_mat3[1:3]
gs_mat5[1:3]
```

## 4. Fisher's exact test

- On each contingency table, we will run Fisher's exact test. 

- We then apply FDR correction to adjust the raw *p*-values for multiple 
testing. 

```{r}
fisher_res <- gs_mat %>% lapply(function(x){
  x %>% fisher.test()
})

fisher_res3 <- gs_mat3 %>% lapply(function(x){
  x %>% fisher.test()
})

fisher_res5 <- gs_mat5 %>% lapply(function(x){
  x %>% fisher.test()
})

fisher_res_p <- fisher_res %>% lapply(function(x){x$p.value})
fisher_res_p3 <- fisher_res3 %>% lapply(function(x){x$p.value})
fisher_res_p5 <- fisher_res5 %>% lapply(function(x){x$p.value})

gs %<>% 
  mutate(fisher_p = fisher_res_p%>%unlist%>%unname,
         fisher_p_3 = fisher_res_p3%>%unlist%>%unname,
         fisher_p_5 = fisher_res_p5%>%unlist%>%unname) %>%
  mutate(fdr = p.adjust(fisher_p, "fdr"),
         fdr_3 = p.adjust(fisher_p_3, "fdr"),
         fdr_5 = p.adjust(fisher_p_5, "fdr"))

```


## 5. Calculate Expected and Observed

- For each gene set, we will calculate the number of genes **expected** to have 
IREs (based on the background proportion) and **observed** value. This 
information will be added to the `gs` object.

```{r }
gs %<>% rowwise() %>% mutate(
  exp_allIRE = (universe_with_ire / universe_without_ire)*n_without_ire,
  obs_allIRE = n_with_ire,
  obs_greater_than_exp_allIRE = obs_allIRE > exp_allIRE,
  
  exp_ire3 = (universe_with_ire3 / universe_without_ire3)*n_without_ire3,
  obs_ire3 = n_with_ire3,
  obs_greater_than_exp_ire3 = obs_ire3 > exp_ire3,
  
  exp_ire5 = (universe_with_ire5 / universe_without_ire5)*n_without_ire5,
  obs_ire5 = n_with_ire5,
  obs_greater_than_exp_ire5 = obs_ire5 > exp_ire5
) 
```


## 6. Results

- The MSigDB gene sets with significant overrepresentation of zebrafish IRE 
gene sets are shown below. 

```{r}
sigGs <- gs %>% dplyr::arrange(fisher_p) %>% 
  dplyr::filter(fdr < 0.05 | fdr_3 < 0.05 | fdr_5 < 0.05) %>%
  dplyr::filter(obs_greater_than_exp_allIRE == TRUE | 
                  obs_greater_than_exp_ire3 == TRUE | 
                  obs_greater_than_exp_ire5 == TRUE) %>%
  dplyr::select(geneset, source, fdr, fdr_3, fdr_5) 
```

- There are `r nrow(sigGs)` gene sets that pass the FDR-adjusted p-value < 
0.05 threshold for significance. 

- The following table shows the top 50 gene sets enriched in IRE genesets, 
ranked by Fisher's exact test *p*-values:

```{r}
gs %>% ungroup() %>%
  arrange(fisher_p) %>%
  dplyr::select(geneset, contains("fdr"), n_with_ire, n, 
                starts_with("obs_greater")) %>% 
  head(50)
```

- Sorted by genesets most enriched in 3' IRE genes:

```{r}
gs %>% ungroup %>% arrange(fisher_p_3) %>% 
  dplyr::select(geneset, contains("fdr"), n_with_ire3, n, 
                starts_with("obs_greater")) %>% head(50)
```

- Sorted by genesets most enriched in 5' IRE genes:
```{r}
gs %>% ungroup %>% arrange(fisher_p_5) %>% 
  dplyr::select(geneset, contains("fdr"), n_with_ire5, n, 
                starts_with("obs_greater")) %>% head(50)
```


## 7. Visualisation (Stacked bar chart)

- The following stacked bar chart shows the overlap between the top 15
genesets and the predicted-IRE genesets.

```{r}
overlapDf <- gs %>% dplyr::arrange(fisher_p) %>% 
  #dplyr::filter(fdr < 0.01 | fdr_3 < 0.01 | fdr_5 < 0.01) %>%
  dplyr::slice(1:15) %>%
  dplyr::filter(obs_greater_than_exp_allIRE == TRUE | obs_greater_than_exp_ire3 == TRUE | obs_greater_than_exp_ire5 == TRUE) %>%
  dplyr::select(geneset, source, n_with_ire3, n_with_ire5, n_without_ire) %>%
  bind_rows(data.frame(
    geneset = c("- All Predicted IREs"),
    source = c("sires"),
    n_with_ire3 = c(length(ireGenes$ire3_all)),
#                    ireUtr3 %>% as.data.frame %>% dplyr::filter(quality == "High") %>% use_series("gene_id") %>% unique %>% length),
    n_with_ire5 = c(length(ireGenes$ire5_all)),
#                    ireUtr5 %>% as.data.frame %>% dplyr::filter(quality == "High") %>% use_series("gene_id") %>% unique %>% length),
    n_without_ire = c(0)
  )) %>% dplyr::mutate(geneset = paste0(geneset, " ", source)) %>%
  dplyr::mutate(geneset = gsub(x=geneset, pattern = "_", replacement = " ")) %>%
  dplyr::select(-source) %>% melt

totals <- overlapDf %>% 
  dplyr::arrange(geneset) %>% 
  group_by(geneset) %>% 
  summarise(sum = sum(value)) %>% 
  arrange(desc(sum)) %>%
  dplyr::filter(geneset != "- All Predicted IREs sires") %>%
  rbind( data.frame(geneset = "All Predicted IREs sires", sum=9999),.)

overlapPlot <- overlapDf %>% 
  left_join(totals) %>%
  arrange(desc(sum)) %>%
  mutate(geneset = factor(geneset, levels = unique(as.character(geneset)))) %>%
  ggplot(aes(x = str_to_title(stringr::str_wrap(geneset, 28)) %>% factor(., levels = unique(as.character(.))), 
                     y = value, 
                     fill = variable)) + 
  geom_col(width = 0.4) +
  theme(aspect.ratio = 1.2, 
        axis.text.x =  element_text(size= 35), 
        axis.title.x = element_text(size= 30),
        axis.title.y = element_text(size = 30), 
        legend.text = element_text(size = 20), 
        legend.title = element_text(size=30),
        axis.text.y = element_text(color = "grey20", size = 13)) + 
  coord_flip() +  # Axis labels too long to be readable so this helps.
  scale_fill_manual(values=c("#e11757", "#3767b3", "#dddddd"), labels =c("3' IRE", "5' IRE", "No IRE")) +
  labs(y = "Number of genes", x = "Gene set", fill = "Gene\nhas\nIRE?") +
  theme_bw()

overlapPlot
```

## 8. Visualisation (Network)

- Although the stacked bar chart shows overlap between different gene sets with 
the predicted IRE geneset, along with information about the size of each geneset,
crucially, it doesn't indicate which genes are in common between the different 
gene sets. My first idea was pairwise Venn diagrams between the IRE genes in
each pair of genesets, but this isn't the most visual / intuitive way to
present the information and it doesn't highlight the genes which are most
often shared between different genesets. Steve had the idea to represent 
this as a network, with distinct groups of genes (grey) representing genesets,
IRE genes in colour, and edges representing overlap between genesets. 


### 8.1. Create the nodes table

I will need to create a `data.frame` with the following columns:

- **Id**: Gene ID
- **IRE**: Factor which can either be `3' IRE`, `5' IRE`, or `no IRE`. 
- **Geneset**: The gene set which it belongs to. 

```{r}
# Include the following top ranked gene sets (~15) in the network 
nodes <- gs %>% arrange(fisher_p) %>% 
  dplyr::filter(fdr < 0.1 | fdr_3 < 0.1 | fdr_5 < 0.1) %>%
  dplyr::filter(obs_greater_than_exp_allIRE == TRUE | 
                  obs_greater_than_exp_ire3 == TRUE | 
                  obs_greater_than_exp_ire5 == TRUE) %>%
  ungroup

# Append information about 3' and 5' predicted IRE genesets 
nodes %<>% bind_rows(
   tibble(
     ids = c(list(ireGenes$ire3_all, ireGenes$ire5_all)),
     geneset = c("Predicted 3' IRE genes", "Predicted 5' IRE genes"),
     source = c("sires","sires"),
     n = c(length(ireGenes$ire3_all), length(ireGenes$ire5_all))
   )
 ) 

# Append the Hallmark Heme Metabolism geneset
# Although this gene set wasnt significantly enriched in 
# IRE-containing genes, it is probably the most comprehensive
# gene set specifically on heme metabolism, and combines info 
# from various studies. 
nodes %<>% bind_rows(
  h_tib %>% filter(geneset == "HALLMARK_HEME_METABOLISM") %>% mutate(n = length(ids[[1]]))
 ) 

nodes
```

First we filtered `gs` for only the genesets that are highly ranked in being 
enriched for predicted IRE genes. We applied the additional filtering step that 
the `observed` number of IRE genes is greater than the `expected` number. This 
results in `r nrow(nodes)` genesets. 

The next step is to create all possible combinations of the genesets. We will 
use the `combn` function to do this. Then we will create a column `common_ids` 
to store the genes which are common to both gene sets being intersected. 

```{r}
gsComb <- combn(nodes$geneset, m = 2) %>% t %>% as.data.frame %>%
  set_colnames(c("geneset", "geneset_2b")) %>%
  left_join(nodes%>%dplyr::select(ids, geneset), by = "geneset") %>%
  dplyr::rename(geneset_nm = geneset,
                geneset = geneset_2b) %>%
  left_join(nodes%>%dplyr::select(ids, geneset), by = "geneset") %>%
  as_tibble %>%
  mutate(common_ids = map2(ids.x, ids.y, ~intersect(.x,.y)))

gsComb 
```

We can extract genes which appear the most often in genesets as follows:

```{r eval=FALSE}
gsComb$common_ids%>%
  unlist %>% 
  table %>% 
  as.data.frame%>%
  arrange(desc(Freq)) %>%
  set_colnames(c("ensembl_gene_id", "f")) %>% 
  left_join(v$genes %>% dplyr::select(-entrezid)) %>% 
  as_tibble
```

The nodes table will contain

- All genes in the genesets
- The geneset names

```{r}
pathways <- nodes$geneset %>% as.data.frame %>% set_colnames("label")
genes <- nodes$ids %>% unlist %>% unique %>% as.data.frame %>% set_colnames("label")
nodesDf <- full_join(pathways, genes, by = "label") %>% rowid_to_column("id")

# nodesDf %>%
#   mutate(text = ifelse(id < 18, label, NA)) %>%
#   mutate(size = ifelse(id < 16, 2, 1)) %>%
#   mutate(colour = ifelse(id < 16, rainbow(26)[id], NA)) 

head(nodesDf,20)

nodesDf2 <- nodesDf %>%
  mutate(
    ire = case_when(
      id < 16 ~ " ",
      id == 18 ~ " ",
      id == 16 ~ "3",
      id == 17 ~ "5",
      label %in% ireGenes$ire3_all ~ "3",
      label %in% ireGenes$ire5_all ~ "5",
      !(label %in% allIREGenes) ~ "no IRE"
    )
  )%>%
  dplyr::select(-id) %>% dplyr::rename(Id = label) 

head(nodesDf2, 30) 
# nodesDf2 %>% write_tsv(here("R","GSEA","data","nodes2.tsv"))
```

Edges will be between:

- The genesets and all genes within the geneset 
- Genes in multiple genesets

```{r}
edgeDf <- nodes$ids %>% set_names(nodes$geneset) %>% plyr::ldply(data.frame) %>% 
  set_colnames(c("pathway", "gene")) %>%
  left_join(nodesDf, by = c("pathway"="label")) %>%
  dplyr::rename(from = id) %>%
  left_join(nodesDf, by = c("gene"="label")) %>%
  dplyr::rename(to=id) %>%
  #dplyr::select(from, to)
  dplyr::select(pathway, gene) %>%
  set_colnames(c("Source","Target"))
edgeDf %>% head(20)


```



## Export Objects

- `gs` object

```{r eval=FALSE}
gs %>% saveRDS(here("data", "ire_ora", "zebrafish","gs.rds"))
```

- MSigDB gene sets with significant over-representation of IRE gene sets

```{r eval=FALSE}
sigGs %>% saveRDS(here("data", "ire_ora", "zebrafish", "sigGs.rds"))
sigGs %>% write_csv(here("data", "ire_ora", "zebrafish", "sigGs.csv"))
```

- Stacked bar chart of the top 15 MSigDB gene sets with overrepresentation of 
IRE gene sets

```{r eval=FALSE}
overlapPlot %>% export::graph2pdf(here("data", "ire_ora", "zebrafish", 
                                       "overlapPlot"))
```

- Edges and nodes tables for plotting in Gephi:

```{r eval=FALSE}
nodesDf2 %>% write_tsv(here("data", "ire_ora", "zebrafish", "nodes.tsv"))
edgeDf %>% write_tsv(here("data", "ire_ora", "zebrafish", "edges.tsv"))
```

