---
title: "Defining IRE Gene Sets - Mouse"
author: "Steve Pederson, Nhi Hin"
date: "2020-04-20"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r load-packages, include=FALSE}
library(BSgenome.Mmusculus.UCSC.mm10)
library(AnnotationHub)
library(plyranges)
library(fgsea)
library(magrittr)
library(scales)
library(pander)
library(tidyverse)
library(here)
```


## 0. Introduction

- Define IRE gene sets for mouse

## 1. Convert Ensembl to UCSC

- In this analysis, the genome used was custom built as BSgenome 
object using the function `BSgenome::forgeBSgenomeDataPkg()`, using the 
2bit file provided as `AH64689` in the package `AnnotationHub()`.


```{r z11}
mm10 <- BSgenome.Mmusculus.UCSC.mm10
ah <- AnnotationHub() %>% subset(species == "Mus musculus")
#query(ah, c("musculus","release-94"))
gr <- ah[["AH64689"]]
```

- The GRanges object containing all annotations for known features 
was then loaded. However, the compatibility of the underlying `seqinfo` 
objects needs to be addressed.

- The seqinfo object of the main genome contains more features, 
and as such will be used as the foundation for the `GRanges` object.

- Whilst the GRanges object is based on Ensembl annotations and chromosome 
identifiers, the genome sequences are based on UCSC chromosome identifiers.

```{r ucscLevels}
ucscLevels <- seqlevels(mm10)
```

- Scaffolds will need to have:
    + the `v` converted to a '.'
    + `chr**_` will need to removed from the prefix
    + `_alt` will need to be removed form the suffix
    
```{r ensLevels}
ensLevels <- ucscLevels %>%
	str_remove_all("^chr") %>%
	str_remove_all("^[0-9Un]+_") %>%
	str_remove_all("_alt") %>%
	str_replace_all("v", ".") %>%
	str_replace_all("^M$", "MT")
```

```{r ens2Ucsc}
ens2Ucsc <- structure(ucscLevels, names = ensLevels)
seqlevels(gr) <- ens2Ucsc[seqlevels(gr)]
seqinfo(gr, new2old = match(seqlevels(mm10), seqlevels(gr))) <- seqinfo(mm10)
```

## 2. Extract UTR sequences

### 2.1. 3' UTR

```{r utr3}
utr3 <- subset(gr, type == "three_prime_utr")
```

```{r exportutr3, eval=FALSE}
utr3Seq <- Views(z11, utr3) %>% DNAStringSet() 
names(utr3Seq) <- paste(utr3$transcript_id, utr3$gene_name, granges(utr3), 
                        sep = "_")
writeXStringSet(utr3Seq, here("data","ire",
                              "mouse","utr3.fa.gz"), compress = TRUE)
```

- A total of `r comma(length(utr3))` sequences were exported


### 2.2. 5' UTR

```{r utr5}
utr5 <- subset(gr, type == "five_prime_utr")
```

```{r exportutr5, eval=FALSE}
utr5Seq <- Views(z11, utr5) %>% DNAStringSet() 
names(utr5Seq) <- paste(utr5$transcript_id, utr5$gene_name, granges(utr5),
                        sep = "_")
writeXStringSet(utr3Seq, here("data","ire",
                              "zebrafish","utr5.fa.gz"), compress = TRUE)
```

- A total of `r comma(length(utr5))` sequences were exported.

## 3. Import SIREs results

#### 3.1. 3' UTR

- Results from scans of the above sequences using the 
[SIREs website](http://ccbg.imppc.org/sires/) were loaded as data frames, 
with co-ordinates converted to the underlying genomic co-ordinates, as opposed 
to being in reference to the submitted sequences

```{r importSireUtr3}
ireUtr3 <- here("data", "ire", "mouse","utr3_sires.gff") %>% 
	import.gff() %>%
	as.data.frame(stringsAsFactors = FALSE) %>%
	as_tibble()
```

```{r utr3ToGRanges}
ireUtr3 %<>%
	mutate(seqnames = as.character(seqnames),
				 seqnames = str_replace(seqnames, "_chrUn_", "_chrUn-")
				 ) %>%
	dplyr::select(-strand) %>%
	separate(seqnames, c("transcript_id", "gene_name", "location"), sep = "_") %>%
	mutate(location = str_replace(location, "chrUn-", "chrUn_")) %>%
	separate(
		location, 
		into = c("seqnames", "trans_start", "trans_end", "strand"),
		sep = "[:-]"
	) %>%
	mutate_at(c("trans_start", "trans_end"), funs(as.integer)) %>%
	mutate(strand = case_when(
		strand == "" ~ "-",
		strand == "+" ~ "+"
	),
	start = ifelse(
		strand == "+", 
		trans_start + start - 1L,
		trans_end - end + 1L
	),
	end = start + width - 1) %>%
	dplyr::select(-contains("trans_")) %>%
  dplyr::filter(!is.na(seqnames))%>%
	makeGRangesFromDataFrame(
		keep.extra.columns = TRUE,
		seqnames.field = "seqnames", 
		start.field = "start", 
		end.field = "end", 
		strand.field = "strand") 
ireUtr3$gene_id <- mcols(gr)$gene_id[match(ireUtr3$transcript_id, gr$transcript_id)]
```

- A total of `r comma(length(ireUtr3))` IREs were found 
across `r comma(length(unique(ireUtr3$gene_id)))` genes when inspecting 
the 3'UTR sequences. These can be summarised as:

```{r summariseUtr3, echo=FALSE}
ireUtr3 %>% 
	mcols() %>% 
	as.data.frame() %>% 
	select(ends_with("class"), quality) %>% 
	lapply(table) %>% 
	pander()
```


### 3.2. 5' UTR

- The same process was repeated for 5' UTR IREs, with the exception that this
time the gene *TMCC1* required manual correcting.

```{r importUtr5}
ireUtr5 <- here("data", "ire", "mouse","utr5_sires.gff") %>% 
	import.gff() %>%
	as.data.frame(stringsAsFactors = FALSE) %>%
	as_tibble()
```

- The loaded matches to IREs in the UTR5 were then converted to a
GRanges object.

```{r utr5ToGRanges}
ireUtr5 %<>%
	mutate(seqnames = as.character(seqnames),
				 seqnames = str_replace(seqnames, "_chrUn_", "_chrUn-")) %>%
	dplyr::select(-strand) %>%
	separate(seqnames, c("transcript_id", "gene_name", "location"), sep = "_") %>%
	mutate(location = str_replace(location, "chrUn-", "chrUn_")) %>%
	separate(
		location, 
		into = c("seqnames", "trans_start", "trans_end", "strand"),
		sep = "[:-]"
	) %>%
	mutate_at(c("trans_start", "trans_end"), funs(as.integer)) %>%
	mutate(strand = case_when(
		strand == "" ~ "-",
		strand == "+" ~ "+"
	),
	start = ifelse(
		strand == "+", 
		trans_start + start - 1L,
		trans_end - end + 1L
	),
	end = start + width - 1) %>%
	dplyr::select(-contains("trans_")) %>%
	makeGRangesFromDataFrame(
		keep.extra.columns = TRUE, 
		seqnames.field = "seqnames", 
		start.field = "start", 
		end.field = "end", 
		strand.field = "strand")
ireUtr5$gene_id <- mcols(gr)$gene_id[match(ireUtr5$transcript_id, gr$transcript_id)]
```

- A total of `r comma(length(ireUtr5))` IREs were found across 
`r comma(length(unique(ireUtr5$gene_id)))` genes when inspecting the
5'UTR sequences. These can be summarised as:

```{r summariseUtr5, echo=FALSE}
ireUtr5 %>% 
	mcols() %>% 
	as.data.frame() %>% 
	select(ends_with("class"), quality) %>% 
	lapply(table) %>% 
	pander()
```

## 5. Genes with both 3' and 5' IREs

- Due to the lack of a simple interpretation for these genes, we will 
remove them from analysis. 

```{r}
ire3and5 <- ireUtr3 %>% 
  dplyr::filter(gene_id %in% ireUtr5$gene_id) %>% 
  as.data.frame %>% 
  dplyr::distinct(gene_id,.keep_all=TRUE) %>%
  dplyr::select(gene_id, seqnames, start, end,
                transcript_id, gene_name, quality) %>%
  dplyr::rename(quality_3ire = quality)

ire5and3 <- ireUtr5 %>% 
  dplyr::filter(gene_id %in% ireUtr3$gene_id) %>% 
  as.data.frame %>% 
  dplyr::distinct(gene_id,.keep_all=TRUE) %>%
  dplyr::select(gene_id, seqnames, start, end,
                transcript_id, gene_name, quality) %>%
  dplyr::rename(quality_5ire = quality)

table(ire3and5$gene_id %in% ire5and3$gene_id) # Check they are the same genes

ire3and5 %<>% full_join(ire5and3, 
                       by = c("gene_id",
                              "seqnames", 
                              "gene_name"), 
                       suffix = c("_3ire", "_5ire")) 

ire3and5
```

- There are `r nrow(ire3and5)` of these genes in total (see above) and overall 
it seems as if the 3' and 5' IREs may be associated with alternative splicing 
in some cases. 


## 4. Defining gene sets

- The gene set definitions are:

    - **3'UTR HQ**: Genes containing high-quality IREs in the 3' UTR
    
    - **3'UTR ALL**: Genes containing any predicted IREs in the 3' UTR 
    (low, med or high quality)
    
    - **5'UTR HQ**: Genes containing high-quality IREs in the 5' UTR

    - **5'UTR ALL**: Genes containing any predicted IREs in the 5' UTR 
    (low, med or high quality)
    
- The genes containing both 3' and 5' IREs will be removed from these 
gene sets. 
    
```{r}
ireGenes <- list(
  ire3_all <- ireUtr3 %>% 
    dplyr::filter(!(gene_id %in% ire3and5$gene_id)) %>% 
    .$gene_id %>%
    unique,
  ire5_all <- ireUtr5%>%
    dplyr::filter(!(gene_id %in% ire3and5$gene_id)) %>% 
    .$gene_id %>%
    unique,
  ire3_hq <- ireUtr3 %>%
    as.data.frame %>% 
    dplyr::filter(quality == "High", 
                  !(gene_id %in% ire3and5$gene_id)) %>% 
    use_series("gene_id") %>% unique,
  ire5_hq <- ireUtr5 %>% 
    as.data.frame %>% 
    dplyr::filter(quality == "High",
                  !(gene_id %in% ire3and5$gene_id)) %>% 
    use_series("gene_id") %>% 
    unique
) %>% set_names(c("ire3_all", "ire5_all", "ire3_hq", "ire5_hq"))

ireGenes %>% str
```



## Export Objects

- The 3' and 5' IRE GRanges (as data.frame objects)

```{r eval=FALSE}
ireUtr3 %>% saveRDS(here("data", "ire", "mouse", "ireUtr3.rds"))
ireUtr5 %>% saveRDS(here("data", "ire", "mouse", "ireUtr5.rds"))
```

- The 3' and 5' IRE gene sets

```{r eval=FALSE}
ireGenes %>% saveRDS(here("data", "ire", "mouse", "ireGenes.rds"))

ireGenes %>% lapply(function(x){
  x %>% 
    as.data.frame() %>%
    set_colnames("ensembl_gene_id")
}) %>% write.xlsx(here("data", "ire", "mouse", "ireGenes.xlsx"))
```

- The genes containing both 3' and 5' IREs

```{r eval=FALSE}
ire3and5 %>% saveRDS(here("data", "ire", "mouse", "ire3and5.rds"))
ire3and5 %>% write.csv(here("data","ire", "mouse", "ire3and5.csv"))
```

